src/tools/kdb/ls.cpp

```diff
@@ -21,12 +21,9 @@ LsCommand::LsCommand () : kdb (root)
 {
 }
 
-int LsCommand::execute (Cmdline const & cl)
+int LsCommand::execute (const Cmdline & cl)
```

Comment:
Okay, its a very strict sense of immutability, which is a bit unusual, e.g. https://isocpp.org/wiki/faq/const-correctness does not recommend it. It is a bit misleading from the interface perspective. But I am okay with it if you use it. (For APIs (like libtools) we should check if all compilers are ABI compatible when `const` needs to be removed).

About the placement of `const`: Anything is okay as long as it is used consistently. It is not so often used in the code base, so even if llvm cannot do it, you can also change it by hand if you prefer one over the other. Otherwise simply use, what is already used.